homeassistant/components/sensor/recorder.py

```diff
@@ -442,6 +449,10 @@ def compile_statistics(  # noqa: C901
                     # The sensor has been reset, update the sum
                     if old_state is not None:
                         _sum += new_state - old_state
+                        sum_increase += sum_increase_tmp
+                        sum_increase_tmp = 0.0
+                        if fstate > 0:
+                            sum_increase_tmp += fstate - 0
```

Comment:
`- 0` ?

Reply:
Ooops, it used to be to make the code more understandable:
```py
                       if fstate > 0:
                            sum_increase_tmp += fstate - 0
                         else:
                            sum_decrease_tmp += 0 - fstate
```

It now serves to achieve the opposite 😓